src/main/python/pybuilder/plugins/python/coverage_plugin.py

```diff
@@ -316,7 +316,14 @@ def _delete_non_essential_modules():
     for module_name in list(sys.modules.keys()):
         module = sys.modules[module_name]
         if module:
-            if not _is_module_essential(module.__name__, sys_packages, sys_modules):
+            try:
+                # if we try to remove non-module object
+                module__name__ = module.__name__
```

Comment:
Hi.
sys.modules can contains specific objects indead of module. For example, pytest uses AliasModule for this purpose (I see the same behaviour in the couple projects. Unfortunately, I still do not understand reason of it)

```
print module
<AliasModule 'py.error' for 'py._error.error'>
print type(module)
<class 'py._apipkg.AliasModule'>
```

In this case module.__name__ will return None.